doc/user-guide/design.rst

```diff
@@ -50,7 +50,7 @@ To see which data is stored for each component, see :doc:`/user-guide/components
 Static data
 -----------
 
-For each component type (line, transformer, generator, etc.), which must be
+For each component variant (line, transformer, generator, etc.), which must be
```

Comment:
I see where this comes from (as `class` and `type` are Python jargon), but do you think it's feasible/sensible to name it "component type" or "component class"?